### PR TITLE
Safer org creation through the controller

### PR DESCRIPTION
**Pull request by ryanpbrewster**
The controller now uses the user request to get the user id instead of
trusting the user to provide it

@Colin-Lane @DerekBurch @jaredpetker @leogrim

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileOrganizationController.scala [hunk 1]
```diff
@@ -27,9 +28,14 @@ class MobileOrganizationController @Inject() (
   // getOrganizationCard
 
   def createOrganization = UserAction.async(parse.tolerantJson) { request =>
-    request.body.validate[OrganizationCreateRequest] match {
-      case _: JsError => Future.successful(BadRequest)
-      case JsSuccess(createRequest, _) =>
+    println("request.body = " + request.body)
+    (request.body \ "name").validate[String] match {
+      case _: JsError =>
+        println("Couldn't find a name in the json")
```
Comment (Colin-Lane) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileOrganizationController.scala:34]: :-1:

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileOrganizationController.scala [hunk 2]
```diff
@@ -27,9 +28,14 @@ class MobileOrganizationController @Inject() (
...
+        Future.successful(BadRequest)
+      case JsSuccess(orgName, _) =>
+        val orgDescriptionOpt = (request.body \ "description").asOpt[String]
+        val createRequest = OrganizationCreateRequest(requesterId = request.userId, orgName, orgDescriptionOpt)
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileOrganizationController.scala:38]: Do we check somewhere up the chain that these Strings are `nonEmpty` ?

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileOrganizationController.scala (hunk 1 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileOrganizationController.scala:34]: Avoid `println`s, they go to the wrong log file. Also, include some context in log messages, like what the `request.body` is

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileOrganizationController.scala [hunk 3]
```diff
@@ -27,9 +28,14 @@ class MobileOrganizationController @Inject() (
   // getOrganizationCard
 
   def createOrganization = UserAction.async(parse.tolerantJson) { request =>
-    request.body.validate[OrganizationCreateRequest] match {
-      case _: JsError => Future.successful(BadRequest)
-      case JsSuccess(createRequest, _) =>
+    println("request.body = " + request.body)
+    (request.body \ "name").validate[String] match {
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileOrganizationController.scala:32]: You can do `(request.body \ "name").asOpt[String].map { ... }`

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileOrganizationController.scala (hunk 2 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/mobile/MobileOrganizationController.scala:38]: Probably fine, the commander can handle that, right? Or would you prefer data validation before?

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala [hunk 4]
```diff
@@ -56,11 +56,11 @@ class OrganizationCommanderImpl @Inject() (
 
   def isValidRequest(request: OrganizationRequest)(implicit session: RSession): Boolean = {
     request match {
-      case OrganizationCreateRequest(createrId, orgName) => true // TODO: can we check ahead of time that an org handle is available?
-      case OrganizationModifyRequest(orgId, requesterId, modifications) =>
+      case OrganizationCreateRequest(createrId, _, _) => true // TODO: can we check ahead of time that an org handle is available?
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala:59]: Yes you can (see `HandleCommander.canBeClaimed`) 

I can actually see the front-end sending the handle on each keystroke while the backend responds on whether it's valid or not - before the user actually submits the organization. @jaredpetker @andrewconner

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala:59]: `canBeClaimed` applies to a specific handle. The commander is actually using `autoSetOrganizationHandle`, and I would need to check whether or not _that_ will succeed. You've said before that it's almost guaranteed to succeed, but I'm going to keep the `try/catch` in for now.

Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationCommander.scala:59]: Oh right.

**Pull request merged**